### ISSUE TITLE
Get Clinicial Report Data - API

### ISSUE THREAD

**Issue by YouriVerveld** [meddbase/meddbase-apis#11]
We would like to show the participants their progress based on the questionnaires that are filled out. These questionnaires are completed by the participants and sent to the clinician. Then, the clinicians have the option to adjust the scores if necessary. Afterward, we would like to receive these scores. Therefore, we cannot use the scores from the completed questionnaires, as they may still be changed by clinicians.

There is mention of a 'Clinical Report.' However, we cannot find this data in the API documentation.

Questions:
1. Is it possible to receive this data via a standard API?
2. Which API in the documentation do we need to use to receive this data?
3. What information do we need to make this API call?

Thank you in advance.

**Comment by meddbase-steve** [comment]
When using the API, there are often requirements in the configuration of the client's Meddbase chamber and the design of workflows and implementation details. This question is likely going to have to be answered in concert with the client and with support from Meddbase staff.

Using the Portal API it may be possible to retrieve information about numeric data recorded in a patient's medical record via:

https://meddbase.github.io/meddbase-apis/patient-portal-api/medical-history/medical-history.html

This data is stored as "Patient Data Fields" in Meddbase, and they can be configured as "Shared with patient" to make them visible via the API.

This might be an area to investigate, if the client you are working with has access to a demo environment and is able to set this up for you.

**Comment by dkevey** [comment]
thanks @meddbase-steve using the method described above (also noted here: https://helpdesk.meddbase.com/hc/en-us/articles/360004610593-Patient-Data-Fields) we can now see the patient total scores from our clinical forms which are saved in the "Patient data fields". If we wanted to included ALL the values from each of our clinical forms, rather than just the total score of each clinical form - is this possible and how do we surface in the API? We can't see that there is an endpoint to view clinical forms. 

As way of background our questionnaires are mapped back to clinical forms which are the "source of truth" and we wish to display the updated values of all fields in our portal if possible please. Hope this makes sense.

**Comment by meddbase-steve** [comment]
Unfortunately the Portal API does not support exposing clinical form data in this way, and I'm not aware of any plans to make that change.

An alternative you could consider is to use native Meddbase functionality to generate an HTML document. That can then be accessed via the API as a PDF.

**Comment by dkevey** [comment]
@meddbase-steve Yes!!! A HTML document (of the clinical form) that can be access via API as a PDF is exactly what we are after. We can see this in the API for questionnaires, but how do we do this for the clinical forms? Can you please point me to the help docs as I couldn't locate it. Thanks

**Comment by meddbase-steve** [comment]
The document templating feature in Meddbase is covered [here](https://helpdesk.meddbase.com/hc/en-us/articles/360000384234-Templates-Documents-An-overview).

A clinical form can be configured with a button to generate a document from a template, as documented [here](https://helpdesk.meddbase.com/hc/en-us/articles/10574691943325-Clinical-Form-Action-Buttons).

Once a document exists, it can be shared with the patient by the clinician ticking the "Share with patient" check box.

It will then be visible via the portal API as a [document](https://meddbase.github.io/meddbase-apis/patient-portal-api/documents/getdocuments.html).
